core/api/views/country_programme.py

```diff
@@ -40,8 +40,60 @@ class CPReportListView(mixins.ListModelMixin, generics.GenericAPIView):
     filterset_class = CPReportFilter
     serializer_class = CPReportSerializer
 
-    def get(self, request, *args, **kwargs):
-        return self.list(request, *args, **kwargs)
+    def _create_cp_records(self, cp_report, section_data, section):
+        for record in section_data:
+            record["country_programme_report_id"] = cp_report.id
+            record["section"] = section
+            record_serializer = CPRecordSerializer(data=record)
+            record_serializer.is_valid(raise_exception=True)
+            record_serializer.save()
+
+    def _create_prices(self, cp_report, section_data):
+        for price in section_data:
+            price["country_programme_report_id"] = cp_report.id
+            price_serializer = CPPricesSerializer(data=price)
+            price_serializer.is_valid(raise_exception=True)
+            price_serializer.save()
+
+    def _create_generation(self, cp_report, section_data):
+        for generation in section_data:
+            generation["country_programme_report_id"] = cp_report.id
+            generation_serializer = CPGenerationSerializer(data=generation)
+            generation_serializer.is_valid(raise_exception=True)
+            generation_serializer.save()
+
+    def _create_emission(self, cp_report, section_data):
+        for emission in section_data:
+            emission["country_programme_report_id"] = cp_report.id
+            emission_serializer = CPEmissionSerializer(data=emission)
+            emission_serializer.is_valid(raise_exception=True)
+            emission_serializer.save()
+
+    def _add_remarks(self, cp_report, section_data):
+        cp_report.comment = section_data.get("remarks", "")
+        cp_report.save()
+
+    def post(self, request, *args, **kwargs):
+        # create cp report
+        cp_report_data = {
+            "name": request.data.get("name"),
+            "year": request.data.get("year"),
+            "country_id": request.data.get("country_id"),
+        }
+        cp_report_serializer = CPReportSerializer(data=cp_report_data)
+        cp_report_serializer.is_valid(raise_exception=True)
+        cp_report = cp_report_serializer.save()
+
+        # create records
+        self._create_cp_records(cp_report, request.data.get("section_a", []), "A")
+        self._create_cp_records(cp_report, request.data.get("section_b", []), "B")
+        self._create_prices(cp_report, request.data.get("section_c", []))
+        self._create_generation(cp_report, request.data.get("section_d", []))
+        self._create_emission(cp_report, request.data.get("section_e", []))
+        self._add_remarks(cp_report, request.data.get("section_f", {}))
+
+        response = CPReportSerializer(cp_report).data
+        return Response(response, status=status.HTTP_200_OK)
```

Comment:
This should likely be in a transaction, as it performs many operations in the DB. 

----

Depending on HOW many operations need to performed here, we may run into the issues of timeout of the request. As performing many DB operations has a significant overhead of sending and receiving data from the DB. 

If there are many such parts, we may consider using bulk_creates instead where possible. This will perform the insert in batches and hence only require one DB operation for many inserts. 

Up to you, I don't really know how the real data looks here.